pkg/kafka/partition/reader.go

```diff
@@ -103,6 +103,9 @@ func (p *Reader) start(ctx context.Context) error {
 
 	// We manage our commits manually, so we must fetch the last offset for our consumer group to find out where to read from.
 	lastCommittedOffset := p.fetchLastCommittedOffset(ctx)
+	if lastCommittedOffset > 0 {
+		lastCommittedOffset++ // We want to begin to read from the next offset, but only if we've previously committed an offset.
```

Comment:
I checked Mimir and I can see that they do this too! 👍 

https://github.com/grafana/mimir/blob/874c48fae80115cf44bab559acaa6ddcabd5317b/pkg/storage/ingest/reader.go#L678

Reply:
How does this compare with the comment on Line 379:

```
lastProducedOffset = lastProducedOffset - 1 // Kafka returns the next empty offset so we must subtract 1 to get the oldest written offset.
```

Reply:
If Kafka returns the next empty offset, why do we need to add 1?

Reply:
That's the last produced offset, vs last committed offset. Apparently they behave slightly differently!